### PR TITLE
MSC2557: Proposal to clarify spoilers

### PR DESCRIPTION
**Pull request by turt2live**
Original MSC: https://github.com/matrix-org/matrix-doc/pull/2010

*Per the proposal process, MSC2010 has been modified to match this MSC.*

[Rendered](https://github.com/matrix-org/matrix-doc/blob/travis/msc/spoiler-fallback/proposals/2557-spoiler-clarifications.md)

### CODE REVIEW

File: proposals/2557-spoiler-clarifications.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+should make reasonable efforts to represent the spoiler in the `body` field of a message.
+
+The recommended fallback format is unchanged.
+
+Additionally, this proposal opens up spoilers to any HTML-supporting message types. Currently
+this includes `m.text` (already included by MSC2010), `m.notice`, and `m.emote`.
+
+## Potential issues
+
+Clients could inadvertadly spoil parts of a message by not representing the spoiler correctly
```
Comment (uhoreg) [inline proposals/2557-spoiler-clarifications.md:18]:
```suggestion
Clients could inadvertently spoil parts of a message by not representing the spoiler correctly
```

Comment (turt2live) [reply proposals/2557-spoiler-clarifications.md:18]: well that's an ironic accident.

File: proposals/2557-spoiler-clarifications.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+## Proposal
+
+The fallback for spoiler messages is optional, though recommended to be sent by clients. Clients
+should make reasonable efforts to represent the spoiler in the `body` field of a message.
+
+The recommended fallback format is unchanged.
+
+Additionally, this proposal opens up spoilers to any HTML-supporting message types. Currently
+this includes `m.text` (already included by MSC2010), `m.notice`, and `m.emote`.
```
Comment (Sorunome) [inline proposals/2557-spoiler-clarifications.md:14]: With https://github.com/matrix-org/matrix-doc/pull/2241 that would also include `m.key.verification.request`

Comment (turt2live) [reply proposals/2557-spoiler-clarifications.md:14]: No? It doesn't require HTML.

Comment (turt2live) [reply proposals/2557-spoiler-clarifications.md:14]: Ah, it does it's just buried in the MSC. That's probably fine still - "don't be silly" clauses start to take effect in the spec.

Comment (Sorunome) [reply proposals/2557-spoiler-clarifications.md:14]: It doesn't require it, but it is optional. Although, as you said "HTML-supporting message types" that would be covered, too

**Comment by turt2live** [comment]
I'm not super convinced this needs an implementation proof given the limited complexity and obvious benefits it provides.

@mscbot fcp merge

**Comment by turt2live** [comment]
Merged 🎉

**Pull request merged**